### PR TITLE
Remove model partition code and refactor model id generation

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are inter-file references, I split a big PR into multiple smaller PRs based on files to save time (1.1 cut off is due this Friday). The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big PR in the end and merge once after all review PRs get approved. For the complete code, check https://github.com/kaituo/anomaly-detection-1/tree/thresholdRCF2

### Description
This PR removed model partitioner as we don’t partition models any more. Also, the PR moved model id logic from ModelPartitioner/ModelManager to ModelIdMapper so we have a central place to single-stream detectors’ id generation.

Testing done:
1. Manually tested single-stream detectors run fine.
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/ml/ModelIdMapper.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+package org.opensearch.ad.ml;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities to map between single-stream models and ids.  We will have circuilar
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ad/ml/ModelIdMapper.java:19]: This class only used for single-flow detector models? If that's the case, how about rename the class as like "SingleFlowModelIdMapper" ?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/ml/ModelIdMapper.java:19]: renamed

File: src/main/java/org/opensearch/ad/cluster/ADClusterEventListener.java [hunk 2]
```diff
@@ -143,7 +144,7 @@ public void clusterChanged(ClusterChangedEvent event) {
                                     LOG.info(REMOVE_MODEL_MSG + " {}", modelId);
                                     modelManager
                                         .stopModel(
-                                            modelManager.getDetectorIdForModelId(modelId),
+                                            SingleStreamModelIdMapper.getDetectorIdForModelId(modelId),
```
Comment (ohltyler) [inline src/main/java/org/opensearch/ad/cluster/ADClusterEventListener.java:147]: Looks like this method was moved from `ModelManager` to `SingleStreamModelIdMapper`, but don't see it removed in this PR. Is it removed in a different PR?

Comment (ohltyler) [reply src/main/java/org/opensearch/ad/cluster/ADClusterEventListener.java:147]: And can you confirm that the usages of the methods moved to `SingleStreamModelIdMapper` have been updated in a different PR?

File: src/main/java/org/opensearch/ad/ml/SingleStreamModelIdMapper.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+package org.opensearch.ad.ml;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities to map between single-stream models and ids.  We will have circuilar
```
Comment (ohltyler) [inline src/main/java/org/opensearch/ad/ml/SingleStreamModelIdMapper.java:19]: nit: `circuilar` -> `circular`

File: src/main/java/org/opensearch/ad/cluster/ADClusterEventListener.java (hunk 2 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/cluster/ADClusterEventListener.java:147]: yes, it is updated in https://github.com/opensearch-project/anomaly-detection/pull/221

**Review by ylwu-amzn — APPROVED** [review]

**Review by ohltyler — APPROVED** [review]

**Pull request closed without merging**